webui/react/src/components/UserAvatar.tsx

```diff
@@ -3,19 +3,37 @@ import React from 'react';
 import Avatar, { Props as AvatarProps } from 'shared/components/Avatar';
 import useUI from 'shared/contexts/stores/UI';
 import { useUsers } from 'stores/users';
+import { DetailedUser } from 'types';
 import { Loadable } from 'utils/loadable';
 import { getDisplayName } from 'utils/user';
 
+import css from './UserAvatar.module.scss';
 export interface Props extends Omit<AvatarProps, 'darkLight' | 'displayName'> {
+  compact?: boolean;
+  table?: boolean;
+  user?: DetailedUser;
   userId?: number;
 }
 
-const UserAvatar: React.FC<Props> = ({ userId, ...rest }) => {
+const UserAvatar: React.FC<Props> = ({ userId, table, user, compact, ...rest }) => {
   const users = Loadable.getOrElse([], useUsers()); // TODO: handle loading state
   const { ui } = useUI();
-  const displayName = getDisplayName(users.find((user) => user.id === userId));
+  const u = user ? user : users.find((user) => user.id === userId);
+  const displayName = getDisplayName(u);
 
-  return <Avatar {...rest} darkLight={ui.darkLight} displayName={displayName} />;
+  const avatar = <Avatar {...rest} darkLight={ui.darkLight} displayName={displayName} />;
+  if (!table || !u) return avatar;
```

Comment:
Instead of two booleans to mean three options, would you support something like Badge's `BadgeType` ? There would be three options for an AvatarType: Table, CompactTable, and Icon?